src/package_manifest.rs

```diff
@@ -88,35 +88,34 @@ impl AsRef<PackageManifest> for PackageManifest {
 }
 
 impl PackageManifest {
-    pub fn get_internal_dependencies<'a, T>(
+    pub fn internal_dependencies_iter<'a, T>(
         &'a self,
-        package_manifests_by_package_name: &HashMap<String, &'a T>,
-    ) -> Vec<&'a PackageManifest>
+        package_manifests_by_package_name: &'a HashMap<String, &'a T>,
+    ) -> impl Iterator<Item = &'a PackageManifest>
     where
         T: AsRef<PackageManifest>,
     {
-        let get_dependency_group = |dependency_group: &str| -> Vec<&'a String> {
-            self.contents
-                .extra_fields
-                .get(dependency_group)
-                .and_then(serde_json::Value::as_object)
-                .map(|object| object.keys().collect())
-                .unwrap_or_default()
-        };
-
-        get_dependency_group("dependencies")
+        static DEPENDENCY_GROUPS: &[&str] = &[
+            "dependencies",
+            "devDependencies",
+            "optionalDependencies",
+            "peerDependencies",
+        ];
+
+        DEPENDENCY_GROUPS
             .iter()
-            .chain(get_dependency_group("devDependencies").iter())
-            .chain(get_dependency_group("optionalDependencies").iter())
-            .chain(get_dependency_group("peerDependencies").iter())
+            // only iterate over the objects corresponding to each dependency group
+            .filter_map(|dependency_group| {
+                self.contents.extra_fields.get(dependency_group)?.as_object()
+            })
+            // get all dependency names from all groups
+            .flat_map(|dependency_group_value| dependency_group_value.keys())
             // filter out external packages
             .filter_map(|package_name| {
                 package_manifests_by_package_name
-                    .get(*package_name)
-                    .cloned()
-                    .map(|thing| thing.as_ref())
+                    .get(package_name)
+                    .map(|&thing| thing.as_ref())
             })
-            .collect()
```

Comment:
Would you call this a performance improvement?

Reply:
I would, as the `collect()` says that we should finish all iteration and store it into a Vec, but every place we use this return value, we just immediately iterate through it. No reason to allocate in that case.

Because Rust iterators are "lazy" (e.g, all allocation or work happens *when you get an element from it*, as opposed to upfront), we also end up allocating significantly less in the long term.

Reply:
Aha, as I suspected :slightly_smiling_face: we can use the `perf` conventional-commit type here. I _hope_ that indicates to semantic-release that we should publish a new version (where `chore` and `refactor` do not)

Reply:
Yes, looks like it does! https://github.com/semantic-release/commit-analyzer#default-rules-matching

Do you mind changing this commit message type, so CI will automatically publish a new version to npm? Else we'll be sleeping on these changes for a while :zzz:

Reply:
Sure thing.